kernel/classes/datatypes/ezuser/ezuser.php

```diff
@@ -1423,8 +1423,8 @@ private static function generateUserCacheData( $userId )
         $data['groups'] = $user->groups();
 
         // role list (session: eZRoleIDList)
-        $groups[] = $userId;
-        $data['roles'] = eZRole::fetchIDListByUser( $groups );
+        $data['groups'][] = $userId;
```

Comment:
This method returns `$data` as result. In original behaviour  `$data['groups']` wasn't contained `$userId` at the end. So I think that this code should looks like as follows:

```php
$groups = $user->groups(); 
$data['groups'] = $groups;
$groups[] = $userId;

$data['roles'] = eZRole::fetchIDListByUser($groups);
```